app/src/main/java/de/adorsys/android/summerparty/data/callback/CartButtonCallback.kt

```diff
@@ -0,0 +1,13 @@
+package de.adorsys.android.summerparty.data.callback
+
+import android.content.Intent
+import android.support.v4.content.ContextCompat.startActivity
+import android.view.View
+import de.adorsys.android.summerparty.ui.CartActivity
+
+class CartButtonCallback : View.OnClickListener {
+
+    override fun onClick(view: View) {
+        startActivity(view.context, Intent(view.context, CartActivity::class.java), null)
```

Comment:
ich würde den als private inner class in die MainActivity packen, damit man direkt den Context der MainActivity referenzierten kann und auch Dinge in das Bundle legen kann, die global in der MainActivity gehalten werden